core/utils/bits.h

```diff
@@ -163,6 +163,24 @@ static inline void MaskBytes(uint8_t *buf, uint8_t const *mask,
   }
 }
 
+// TODO(melvinw): it would be nice if MaskTopN() supported blobs.
+// Returns a mask for the first `n` bits of an integral type T, i.e. starting
+// from the most significant bit toward the least significant.
+// For example: MaskTopN<uint32_t>(9) will return 0xF8000000
+template <typename T, typename = std::enable_if<std::is_integral<T>::value>>
+static inline T MaskTopN(size_t n) {
+  return (n == sizeof(T) * 8) ? ~T{0} : (T{1} << n) - 1;
```

Comment:
Question: should this use >= here, in which case, we don't need the other test in `ip.cc`? It's not clear to me what it means to ask for the top 47 bits in a 32 bit mask...

Reply:
Good point! Fixed